lib/chewy/fields/base.rb

```diff
@@ -1,19 +1,24 @@
 module Chewy
   module Fields
     class Base
-      attr_reader :name, :options, :value, :children
+      attr_reader :name, :options, :children
       attr_accessor :parent
 
+      JOIN_FIELD_EXTRA_OPTIONS = %i[join_id join_type].freeze
+
       def initialize(name, value: nil, **options)
         @name = name.to_sym
         @options = {}
         update_options!(**options)
         @value = value
         @children = []
+        @allowed_relations = find_allowed_relations(options[:relations]) # for join fields
```

Comment:
Why do we treat `relations`, `join_id` & `join_type` in different manner?

Reply:
`relations` will be passed to elastic to define the parent-children hierarchy. `join_id` and `join_type` are just a syntactic sugar that allow us to define `value` correctly, they're not passed to elastic.